src/index.test.ts

```diff
@@ -1,153 +1,238 @@
   });
 
-  test('dns.lookup override works with callback', (done) => {
+  test("dns.lookup override works with callback", (done) => {
     manager.initialize();
-    dns.lookup('test-jest.local', (err, address) => {
+    dns.lookup("test-jest.local", (err, address) => {
       expect(err).toBeNull();
-      expect(['1.2.3.4', '2001:db8::1']).toContainEqual(address);
+      expect(["1.2.3.4", "2001:db8::1"]).toContainEqual(address);
       done();
     });
   });
 
-  test('dns.lookup override works with promises', async () => {
-    const result = await (dns.lookup as any).__promisify__('test-jest.local');
-    expect(['1.2.3.4', '2001:db8::1']).toContainEqual(result.address);
+  test("dns.lookup override works with promises", async () => {
+    const result = await (dns.lookup as any).__promisify__("test-jest.local");
+    expect(["1.2.3.4", "2001:db8::1"]).toContainEqual(result.address);
   });
 
-  test('updateCache merges entries', async () => {
-    await manager.updateCache('custom.local', {
-      addresses: { ipv4: ['1.1.1.1'], ipv6: [] },
-      activeAddress: '1.1.1.1',
+  test("updateCache merges entries", async () => {
+    await manager.updateCache("custom.local", {
+      addresses: { ipv4: ["1.1.1.1"], ipv6: [] },
+      activeAddress: "1.1.1.1",
       family: 4,
       timestamp: Date.now(),
       ttl: 60000,
       hits: 0,
-      misses: 0
+      misses: 0,
     });
-    const entry = manager.getCacheEntry('custom.local');
-    expect(entry?.activeAddress).toBe('1.1.1.1');
+    const entry = manager.getCacheEntry("custom.local");
+    expect(entry?.activeAddress).toBe("1.1.1.1");
   });
 
-  test('clearHostname removes entry', async () => {
+  test("clearHostname removes entry", async () => {
     // Test clearing a specific hostname
-    await manager.lookup('test-jest.local');
-    manager.clearHostname('test-jest.local');
-    expect(manager.getCacheEntry('test-jest.local')).toBeNull();
+    await manager.lookup("test-jest.local");
+    manager.clearHostname("test-jest.local");
+    expect(manager.getCacheEntry("test-jest.local")).toBeNull();
   });
 
-  test('clear() wipes all stats & cache', () => {
+  test("clear() wipes all stats & cache", () => {
     // Test clearing all cache entries
     manager.clear();
     expect(manager.getStats().size).toBe(0);
   });
+
+  test("dns.lookup honors family:4 (Force IPv4 ON)", (done) => {
+    manager.initialize();
+    dns.lookup("test-jest.local", { family: 4 }, (err, address, family) => {
+      expect(err).toBeNull();
+      expect(address).toBe("1.2.3.4");
+      expect(family).toBe(4);
+      done();
+    });
+  });
+
+  test("dns.lookup default (Force IPv4 OFF) uses cache default (ipv6-first here)", (done) => {
+    dns.lookup("test-jest.local", (err, address, family) => {
+      expect(err).toBeNull();
+      expect(address).toBe("2001:db8::1");
+      expect(family).toBe(6);
+      done();
+    });
+  });
```

Comment:
These 2 tests are quite similar to the previous `switches to IPv4 when forced` and `caches DNS lookups` tests?

Reply:
Ok I got it.
It's testing the same underlying function but since it's not calling it the same way IMO it makes sense to keep it